spec.html

```diff
@@ -7082,17 +7098,18 @@ <h1>OrdinaryOwnPropertyKeys ( _O_ )</h1>
       </emu-clause>
     </emu-clause>
 
-    <emu-clause id="sec-objectcreate" aoid="ObjectCreate">
-      <h1>ObjectCreate ( _proto_ [ , _internalSlotsList_ ] )</h1>
-      <p>The abstract operation ObjectCreate with argument _proto_ (an object or null) is used to specify the runtime creation of new ordinary objects. The optional argument _internalSlotsList_ is a List of the names of additional internal slots that must be defined as part of the object. If the list is not provided, a new empty List is used. This abstract operation performs the following steps:</p>
+    <emu-clause id="sec-OrdinaryObjectCreate" aoid="OrdinaryObjectCreate">
+      <h1>OrdinaryObjectCreate ( _proto_ [ , _additionalInternalSlotsList_ ] )</h1>
+      <p>The abstract operation OrdinaryObjectCreate with argument _proto_ (an object or *null*) is used to specify the runtime creation of new ordinary objects. The optional argument _additionalInternalSlotsList_ is a List of the names of additional internal slots that must be defined as part of the object, beyond [[Prototype]] and [[Extensible]]. If the list is not provided, a new empty List is used. This abstract operation performs the following steps:</p>
       <emu-alg>
-        1. If _internalSlotsList_ is not present, set _internalSlotsList_ to a new empty List.
-        1. Let _obj_ be a newly created object with an internal slot for each name in _internalSlotsList_.
-        1. Set _obj_'s essential internal methods to the default ordinary object definitions specified in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots"></emu-xref>.
-        1. Set _obj_.[[Prototype]] to _proto_.
-        1. Set _obj_.[[Extensible]] to *true*.
-        1. Return _obj_.
+        1. Let _internalSlotsList_ be &laquo; [[Prototype]], [[Extensible]] &raquo;.
+        1. If _additionalInternalSlotsList_ was provided, append all of its items to _internalSlotsList_.
+        1. Return ! AllocateBasicObject(_internalSlotsList_, _proto_).
       </emu-alg>
+
+      <emu-note>
+        <p>Apart from being slightly easier to call than AllocateBasicObject, using OrdinaryObjectCreate communicates the intention to create an ordinary object, and not an exotic one. Thus, it must not be called as part as of an algorithm that subsequently modifies the internal methods of the object in ways that would make the result non-ordinary. In such cases, use AllocateBasicObject. (Even if it means having to include [[Prototype]] and [[Extensible]] in the internal slots list manually.)</p>
```

Comment:
I see that this somewhat covers the point I mentioned about mutating the set of essential internal methods.   I think it is good to say it both places.